homeassistant/components/climate/nest.py

```diff
@@ -168,18 +182,24 @@ def is_away_mode_on(self):
     def set_temperature(self, **kwargs):
         """Set new target temperature."""
         import nest
+        temp = None
         target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
         target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
         if self._mode == NEST_MODE_HEAT_COOL:
             if target_temp_low is not None and target_temp_high is not None:
                 temp = (target_temp_low, target_temp_high)
+                _LOGGER.debug("Nest set_temperature-output-value=%s", temp)
         else:
             temp = kwargs.get(ATTR_TEMPERATURE)
-        _LOGGER.debug("Nest set_temperature-output-value=%s", temp)
+            _LOGGER.debug("Nest set_temperature-output-value=%s", temp)
         try:
-            self.device.target = temp
-        except nest.nest.APIError:
-            _LOGGER.error("An error occurred while setting the temperature")
+            if temp is not None:
+                self.device.target = temp
+        except nest.nest.APIError as api_error:
+            _LOGGER.error("An error occurred while setting temperature: %s",
+                          api_error)
+            # restore target temperature
+            self.hass.add_job(self.async_update_state)
```

Comment:
I don't really see why we would have to restore target temperature? Where was it overridden?

Also, just call `hass.schedule_update_ha_state(True)`